File: 8kyu/(8 kyu) String repeat/(8 kyu) String repeat.py
# 1 Plain Solution
def repeat_str(repeat, string):
    str = ""
    
    for i in range(repeat):
        str = str + string
        
    return str

# 2 Optimized solution
def repeat_str(repeat, string):
    return "{}".format(string) * int(repeat)

# 3 Clever solution
def repeat_str(repeat, string):
    return repeat * string

# 4 Coding golf
repeat_str = lambda n, str: str * n

File: 8kyu/(8 kyu) String repeat/(8 kyu) String repeat.java
// #1
// public class Solution {
//     public static String repeatStr(final int repeat, final String string) {
//         String result = "";
//         for (int i = 0; i < repeat; i++) {
//           result += string;
//         }
//         return result;
//     }
// }
// #2
public class Solution {
    public static String repeatStr(final int repeat, final String string) {
        return string.repeat(repeat);
    }
}
